### PR TITLE
Build loki-build-image for multiple architectures.

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
The current `loki-build-image` is for `amd64` only. However, many are developing Loki on Apple Silicon. That's why we should publish build images for `arm64` as well.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: .drone/drone.jsonnet [hunk 1]
```diff
@@ -503,32 +503,39 @@ local manifest_ecr(apps, archs) = pipeline('manifest-ecr') {
...
+        depends_on: ['clone'],
+        volumes: [
+          {
+            name: 'docker',
+            path: '/var/run/docker.sock',
+          },
+        ],
+        privileged: true,
+      },
+      make('build-image-push', container=false) {
```
Comment (jeschkies) [inline .drone/drone.jsonnet:527]: The target needs authentication for Docker.

**Comment by jeschkies** [comment]
To speed things up we should probably push several images with the `-<arch>` suffix and then create the manifest https://github.com/grafana/loki/blob/main/.drone/docker-manifest.tmpl

File: .drone/drone.jsonnet [hunk 2]
```diff
@@ -497,40 +497,58 @@ local manifest_ecr(apps, archs) = pipeline('manifest-ecr') {
...
+          dry_run: false,
+          platform: 'linux/' + arch,
         },
       },
+    ],
+  }
+  for arch in ['amd64', 'arm64']
+] + [
+  pipeline('loki-build-image-publish') {
+    local build_image_tag = '0.32.0-test',
```
Comment (MichelHollands) [inline .drone/drone.jsonnet:530]: Is this still needed? It doesn't seem to be used anymore.

Comment (jeschkies) [reply .drone/drone.jsonnet:530]: You are right. I need to check if this is a bug.

**Comment by jeschkies** [comment]
Closed in favour of https://github.com/grafana/loki/pull/11130

**Pull request closed without merging**